src/acl/group.test.ts

```diff
@@ -0,0 +1,1158 @@
+    )
+  );
+  aclDataset.add(
+    DataFactory.quad(
+      DataFactory.namedNode(subjectIri),
+      DataFactory.namedNode("http://www.w3.org/ns/auth/acl#agentGroup"),
+      DataFactory.namedNode(group)
+    )
+  );
+  if (access.read) {
+    aclDataset.add(
+      DataFactory.quad(
+        DataFactory.namedNode(subjectIri),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#mode"),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#Read")
+      )
+    );
+  }
+  if (access.append) {
+    aclDataset.add(
+      DataFactory.quad(
+        DataFactory.namedNode(subjectIri),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#mode"),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#Append")
+      )
+    );
+  }
+  if (access.write) {
+    aclDataset.add(
+      DataFactory.quad(
+        DataFactory.namedNode(subjectIri),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#mode"),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#Write")
+      )
+    );
+  }
+  if (access.control) {
+    aclDataset.add(
+      DataFactory.quad(
+        DataFactory.namedNode(subjectIri),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#mode"),
+        DataFactory.namedNode("http://www.w3.org/ns/auth/acl#Control")
+      )
+    );
+  }
+
+  return Object.assign(aclDataset, { accessTo: resource });
+}
+
+function addAclDatasetToLitDataset(
+  litDataset: LitDataset & WithResourceInfo,
+  aclDataset: unstable_AclDataset,
+  type: "resource" | "fallback"
+): LitDataset & WithResourceInfo & unstable_WithAcl {
+  const acl: unstable_WithAcl["acl"] = {
+    fallbackAcl: null,
+    resourceAcl: null,
+    ...(((litDataset as any) as unstable_WithAcl).acl ?? {}),
+  };
+  if (type === "resource") {
+    litDataset.resourceInfo.unstable_aclUrl =
+      aclDataset.resourceInfo.fetchedFrom;
+    aclDataset.accessTo = litDataset.resourceInfo.fetchedFrom;
+    acl.resourceAcl = aclDataset;
+  } else if (type === "fallback") {
+    acl.fallbackAcl = aclDataset;
+  }
+  return Object.assign(litDataset, { acl: acl });
+}
+
+function getMockDataset(fetchedFrom: IriString): LitDataset & WithResourceInfo {
+  return Object.assign(dataset(), {
+    resourceInfo: {
+      fetchedFrom: fetchedFrom,
+      isLitDataset: true,
+    },
+  });
+}
+
+describe("getGroupAccessOne", () => {
+  it("returns the Resource's own applicable ACL rules", () => {
+    const litDataset = getMockDataset("https://some.pod/container/resource");
+    const resourceAcl = addAclRuleQuads(
+      getMockDataset("https://some.pod/container/resource.acl"),
+      "https://some.pod/group#webId",
+      "https://some.pod/container/resource",
+      { read: false, append: false, write: false, control: true },
+      "resource"
+    );
+    const litDatasetWithAcl = addAclDatasetToLitDataset(
+      litDataset,
+      resourceAcl,
+      "resource"
+    );
+
+    const access = unstable_getGroupAccessOne(
```

Comment:
Ah this is a good idea, to name it `Group` rather than `AgentGroup`. Note to self: rename `agentClass.ts` to `class.ts`.

Reply:
Refactor express, at your service: #228